gax/src/main/java/com/google/api/gax/rpc/Callables.java

```diff
@@ -56,12 +57,19 @@ private Callables() {}
       UnaryCallSettings<?, ?> callSettings,
       ClientContext clientContext) {
 
-    if (areRetriesDisabled(callSettings.getRetryableCodes(), callSettings.getRetrySettings())) {
-      // When retries are disabled, the total timeout can be treated as the rpc timeout.
-      return innerCallable.withDefaultCallContext(
-          clientContext
-              .getDefaultCallContext()
-              .withTimeout(callSettings.getRetrySettings().getTotalTimeout()));
```

Comment:
Changing this means that non-retryable RPCs will be returned wrapped in a RetryingCallable, which means AttemptCallable will be used for the lone RPC to be made and if a user was previously providing an ApiCallContext.withTimeout, AttemptCallable will [override] it (unless we change AttemptCallable to respect the ApiCallContext.withTimeout).

[override]: https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/rpc/AttemptCallable.java#L72-L75

Changing this here is important, however, because the RetryingCallable is configured by the RetryingAlgorithm which supports the per-call overallTimeout adjustments via the RetryingContext.